### DISCUSSION TITLE
Can I use babylon.js as a 3d engine in tslab?

### DISCUSSION THREAD

**Discussion by Lowclouds** [yunabe/tslab#77] [Q&A] [answered]
I haven't installed tslab yet, but reading the part about the display function left me thinking this wouldn't work, since babylon.js depends on access to a canvas.

**Comment by yunabe** [comment, marked as answer]
Right. At this moment, tslab is a server side JS kernel and we cannot run a 3d engine with client-side JS easily.

I think the support of building client-side JS is an interesting extension of this kernel. But it's not very tirival how to briedge data from the server-side JavaScript to the cilent-side JS in Jupyter notebooks.
